### PR TITLE
Improved user feedback for voting section

### LINKED ISSUE #2405: Update voting section of PNI site
Based on user testing and feedback, we are updating the PNI site vote section to try and reduce user errors/ frustrations. Currently the vote section doesn't provide error feedback if the user only gets through one vote section. 

[Redpen with instructions is here.](https://redpen.io/p/oq8a929059c9e4976b)

Updates:

- [ ] Add error 'hint' state to 'How likely are you to buy it' if the user tries to click 'Vote' button after only completing the 'how creepy is this product'
- [ ] Add hover state to 'How likely are you to buy it'
- [ ] Add clicked state to 'How likely are you to buy it'

P2 if there is time/ if possible:
- [ ] Add a hint (when user scrolls down the page) to the 'How creepy is this product' (as in the gif example below)

02_creep-o-meter-hint

cc @beccaklam @kristinashu @gideonthomas

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #2405 

- [x] Add hover and click state to capsule button `#3bb7ef`
- [x] `How likely are you to buy it?` capsule buttons wiggle if user tries to submit the form without answering that question
- [x] Submit button stays slightly opaque until the two questions have been answered
- [x] Voting functionality is still working

### CODE REVIEW

**Comment by mmmavis** [comment]
@Pomax I'll update the original ticket so that it reflects the new specs we introduced. I got r+ for design for this PR so this is now ready for code review! 👀

**Comment by mmmavis** [comment]
@Pomax i think i broke the voting functionality. Will add you back as reviewer once I fix the bug.

**Comment by mmmavis** [comment]
@Pomax this is ready for review! 😄

**Comment by Pomax** [comment]
Should the user have to at least click the slider before they get to vote? Even if it means changing it to the value it already is?

**Comment by Pomax** [comment]
Closes https://github.com/mozilla/foundation.mozilla.org/issues/2060 as well

File: source/js/buyers-guide/components/creep-vote/creep-vote.jsx [hunk 1]
```diff
@@ -85,6 +86,14 @@ export default class CreepVote extends React.Component {
       });
   }
 
+  handleSubmitBtnClick() {
+    this.setState({
+      submitAttempted: true
+    }, () => {
+      setTimeout(() => this.setState({ submitAttempted: false }), 1000); // The 1s timeout is for the wiggle animation to finish. See &:invalid.submit-attempted ruleset in creep-vote.scss for details
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/js/buyers-guide/components/creep-vote/creep-vote.jsx]: used `animationend` instead

**Comment by mmmavis** [comment]
> Should the user have to at least click the slider before they get to vote? Even if it means changing it to the value it already is?

@Pomax I chatted with @kristinashu - let's not worry about validating/detecting movement for the slider

**Comment by mmmavis** [comment]
@Pomax 👀 ?

File: source/js/buyers-guide/components/creep-vote/creep-vote.jsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import CREEPINESS_LABELS from "../creepiness-labels.js";
 export default class CreepVote extends React.Component {
   constructor(props) {
     super(props);
+    this.wouldbuyBtnGroup = React.createRef();
```
Comment (Pomax) [inline source/js/buyers-guide/components/creep-vote/creep-vote.jsx]: Reach has support for `onAnimationEnd`, so we don't need a ref for this.

File: source/js/buyers-guide/components/creep-vote/creep-vote.jsx [hunk 3]
```diff
@@ -39,14 +40,23 @@ export default class CreepVote extends React.Component {
...
+      submitAttempted: false
     };
   }
 
   componentDidMount() {
     if (this.props.whenLoaded) {
       this.props.whenLoaded();
     }
+
+    if (this.wouldbuyBtnGroup.current) {
```
Comment (Pomax) [inline source/js/buyers-guide/components/creep-vote/creep-vote.jsx]: instead, make this a handler:
```
handleAnimationEnd(evt) {
  if (evt.animationName === `wiggle`) {
    this.setState({
      submitAttempted: false
    });
  }
}
```

File: source/js/buyers-guide/components/creep-vote/creep-vote.jsx [hunk 4]
```diff
@@ -137,7 +153,7 @@ export default class CreepVote extends React.Component {
             <h3 className="h5-heading mb-2">How likely are you to buy it?</h3>
           </div>
           <div className="text-center">
-            <div class="btn-group btn-group-toggle mt-3 mt-md-5" data-toggle="buttons">
+            <div class="btn-group btn-group-toggle mt-3 mt-md-5" data-toggle="buttons" ref={this.wouldbuyBtnGroup}>
```
Comment (Pomax) [inline source/js/buyers-guide/components/creep-vote/creep-vote.jsx]:
```suggestion
            <div class="btn-group btn-group-toggle mt-3 mt-md-5" data-toggle="buttons" onAnimationEnd={evt => this.handleAnimationEnd(evt)}>
```

**Comment by mmmavis** [comment]
🎉

**Pull request merged**